File: app/src/net/reichholf/dreamdroid/helpers/NameValuePair.java
package net.reichholf.dreamdroid.helpers;

import android.text.TextUtils;

import java.io.UnsupportedEncodingException;
import java.net.URLEncoder;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by reichi on 18/08/15.
 */
public class NameValuePair {
	private String mKey;
	private String mValue;
	public NameValuePair(String key, String value)
	{
		mKey = key;
		mValue = value;
	}

	public String key(){
		return mKey;
	}

	public String value() {
		if(mValue == null)
			return "";
		return mValue;
	}

	public static String toString(NameValuePair pair) {
		String value = "";
		try {
			value = URLEncoder.encode(pair.value(), "utf-8").replace("+", "%20");
		} catch (UnsupportedEncodingException e) {
			e.printStackTrace();
		}
		return String.format("%s=%s", pair.key(), value);
	}

	public static String toString(List<NameValuePair> pairs) {
		ArrayList<String> params = new ArrayList<>();
		for(NameValuePair pair : pairs) {
			params.add(toString(pair));
		}
		return TextUtils.join("&", params);
	}
}

File: app/src/net/reichholf/dreamdroid/helpers/NameValuePair.kt
package net.reichholf.dreamdroid.helpers

import java.io.UnsupportedEncodingException
import java.net.URLEncoder

class NameValuePair(val key: String, private val rawValue: String?) {
    fun key(): String = key

    fun value(): String = rawValue ?: ""

    companion object {
        fun toString(pair: NameValuePair): String {
            var value = ""
            try {
                value = URLEncoder.encode(pair.value(), "utf-8").replace("+", "%20")
            } catch (e: UnsupportedEncodingException) {
                e.printStackTrace()
            }
            return String.format("%s=%s", pair.key(), value)
        }

        fun toString(pairs: List<NameValuePair>): String {
            if (pairs.isEmpty()) return ""
            val params = ArrayList<String>(pairs.size)
            for (pair in pairs) {
                params.add(toString(pair))
            }
            return params.joinToString("&")
        }
    }
}
